### PR TITLE
mythen3: virtual server, connected timing mode, row and col in header…

### PR DESCRIPTION
**Pull request by thattil**
…, included pattern bit and mask

### CODE REVIEW

**Comment by thattil** [comment]
additionally, get/set timing, generate virtual data for gotthard2, replacing vref_restore to vref_rstore, burst type to burst mode

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c [hunk 1]
```diff
@@ -1468,14 +1555,22 @@ u_int32_t runBusy() {
...
-	FILE_LOG(logDEBUG1, ("Status Register: %08x\n", s));
+	//FILE_LOG(logDEBUG1, ("Status Register: %08x\n", s));
 	return s;
 }
 
 /* common */
 
 int calculateDataBytes() {
-	return 0;
+	int numCounters = __builtin_popcount(getCounterMask());
```
Comment (erikfrojdh) [inline slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c]: There is no 8 bit mode?

I'm also not sure that this works for 1 bit. Seems to round down for me...

File: slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c [hunk 2]
```diff
@@ -1106,36 +1106,46 @@ int setHighVoltage(int val) {
...
-	enum timingMode ret=GET_TIMING_MODE;
-	if (arg != GET_TIMING_MODE) {
-		switch(arg) {
-		case AUTO_TIMING:			ret = 0;	break;
-		case TRIGGER_EXPOSURE:		ret = 2;	break;
-		case BURST_TRIGGER:			ret = 1;	break;
-		case GATED:					ret = 3;	break;
-		}
-		FILE_LOG(logDEBUG1, ("Setting Triggering Mode: %d\n", (int)ret));
+	int ret = 0;
```
[the remark this answers is not in the document]
Comment (thattil) [reply slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c]: Sure, this is the part that goes to Ian's code. When we rewrite the eiger server, we can fix these things.

[the remark this answers is not in the document]
Comment (thattil) [reply slsDetectorServers/eigerDetectorServer/slsDetectorFunctionList.c]: Sure, this is the part that goes to Ian's code. When we rewrite the eiger server, we can fix these things.

File: slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c (hunk 1 above)
Comment (thattil) [reply slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c]: Nope, only 1, 4 16 and 24. I will look into it once I allow them to change dynamic range in the server.

File: python/src/enums.cpp [hunk 3]
```diff
@@ -169,7 +169,7 @@ void init_enums(py::module &m) {
         .value("VB_COMP_FE", slsDetectorDefs::dacIndex::VB_COMP_FE)
         .value("VB_COMP_ADC", slsDetectorDefs::dacIndex::VB_COMP_ADC)
         .value("VCOM_CDS", slsDetectorDefs::dacIndex::VCOM_CDS)
-        .value("VREF_RESTORE", slsDetectorDefs::dacIndex::VREF_RESTORE)
+        .value("VREF_RSTORE", slsDetectorDefs::dacIndex::VREF_RSTORE)
```
[the remark this answers is not in the document]
Comment (thattil) [reply python/src/enums.cpp]: Next time I will do that, thanks

**Pull request merged**